pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -197,6 +210,22 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
+type ResourceSelector struct {
+	// name of an object within a claimed group/resource.
+	// It matches the metadata.name field of the underlying object.
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern="^([a-z0-9][-a-z0-9_.]*)?[a-z0-9]$"
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+
+	// namespace containing the named object. Matches metadata.namespace field.
+	// If "name" is unset, all objects from the namespace are being claimed.
+	//
+	// +required
+	Namespace string `json:"namespace"`
```

Comment:
why not omit empty?

Reply:
and why not optional?

Reply:
being required does not extend to more selector types.

Reply:
fixed, however how do we want to validate this, using CEL?